toxcore/tox.c

```diff
@@ -547,8 +547,14 @@ Tox *tox_new(const struct Tox_Options *options, Tox_Err_New *error)
         m_options.proxy_info.ip_port.port = net_htons(tox_options_get_proxy_port(opts));
     }
 
-    tox->mono_time = mono_time_new();
-    tox_set_network(tox, nullptr);
+    const Tox_System *sys = tox_options_get_operating_system(opts);
+    const Tox_System default_system = tox_default_system();
+    if (sys == nullptr) {
+        sys = &default_system;
```

Comment:
Why put the function call outside the conditional? Inside would be cleaner IMHO and independent of side effects of `tox_default_system()`

Reply:
That would mean making default_system non-const. Would you prefer splitting the definition and initialisation, making the value non-const and uninitialised by default to avoid side effects of `tox_default_system()`?

Reply:
misread the purpose of this code